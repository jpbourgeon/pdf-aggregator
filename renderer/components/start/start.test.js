const React = require('react');
const ShallowRenderer = require('react-test-renderer/shallow');
const { RenderView } = require('../start');
const { defaultState } = require('../../store');

describe('The renderView component', () => {
  it('should render correctly', () => {
    const props = {
      actions: {
        openDialog: () => jest.fn(),
        getFolder: () => jest.fn(),
        handleChange: () => jest.fn(),
        resetState: () => jest.fn(),
        submit: () => jest.fn(),
      },
      classes: {
        root: jest.fn(),
        title: jest.fn(),
        formControl: jest.fn(),
        smallFormControl: jest.fn(),
        button: jest.fn(),
      },
      parameters: {
        currentWindow: jest.fn(),
        dialogOptions: () => jest.fn(),
      },
      state: defaultState,
    };
    const renderer = new ShallowRenderer();
    renderer.render(<RenderView {...props} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
