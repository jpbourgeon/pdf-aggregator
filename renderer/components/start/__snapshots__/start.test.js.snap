// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The renderView component should render correctly 1`] = `
<div
  className={[MockFunction]}
>
  <WithStyles(GridWrapper)
    container={true}
    spacing={24}
  >
    <WithStyles(GridWrapper)
      item={true}
      xs={12}
    >
      <WithStyles(Typography)
        className={[MockFunction]}
        variant="display1"
      >
        Paramètres
      </WithStyles(Typography)>
      <WithStyles(FormControl)
        className={[MockFunction]}
      >
        <WithStyles(InputLabel)
          htmlFor="input"
          shrink={true}
        >
          Dossier source
        </WithStyles(InputLabel)>
        <WithStyles(Input)
          disabled={true}
          id="input"
          startAdornment={
            <WithStyles(InputAdornment)
              position="start"
            >
              <WithStyles(IconButton)
                aria-label="Choisir le dossier source"
                onClick={[Function]}
              >
                <pure(FolderOpen) />
              </WithStyles(IconButton)>
            </WithStyles(InputAdornment)>
          }
          type="text"
          value=""
        />
      </WithStyles(FormControl)>
      <WithStyles(FormControl)
        className={[MockFunction]}
      >
        <WithStyles(InputLabel)
          htmlFor="output"
          shrink={true}
        >
          Dossier cible
        </WithStyles(InputLabel)>
        <WithStyles(Input)
          disabled={true}
          id="output"
          label=""
          startAdornment={
            <WithStyles(InputAdornment)
              position="start"
            >
              <WithStyles(IconButton)
                aria-label="Choisir le dossier cible"
                onClick={[Function]}
              >
                <pure(FolderOpen) />
              </WithStyles(IconButton)>
            </WithStyles(InputAdornment)>
          }
          type="text"
          value=""
        />
      </WithStyles(FormControl)>
      <WithStyles(FormControl)
        className={[MockFunction]}
      >
        <WithStyles(InputLabel)
          htmlFor="filename"
          shrink={true}
        >
          Nom de(s) fichier(s) cible(s) (options : %dossiersouce%, %dateiso%)
        </WithStyles(InputLabel)>
        <WithStyles(Input)
          id="filename"
          onChange={[Function]}
          type="text"
          value="%dossiersource%_%dateiso%"
        />
      </WithStyles(FormControl)>
      <WithStyles(FormControl)
        className={[MockFunction]}
      >
        <WithStyles(InputLabel)
          htmlFor="title"
          shrink={true}
        >
          Titre de(s) document(s) (options : %dossiersource%, %datefr%, %ligne%)
        </WithStyles(InputLabel)>
        <WithStyles(Input)
          id="title"
          onChange={[Function]}
          type="text"
          value="%dossiersource%%ligne%%datefr%"
        />
      </WithStyles(FormControl)>
      <WithStyles(FormControl)
        className={[MockFunction]}
      >
        <WithStyles(InputLabel)
          htmlFor="level"
          shrink={true}
        >
          Agréger au niveau
        </WithStyles(InputLabel)>
        <WithStyles(Input)
          className={[MockFunction]}
          id="level"
          onChange={[Function]}
          type="number"
          value={1}
        />
      </WithStyles(FormControl)>
      <WithStyles(FormControl)
        className={undefined}
      >
        <WithStyles(FormControlLabel)
          control={
            <WithStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />
          }
          label="Journal des modifications"
        />
        <WithStyles(FormControlLabel)
          control={
            <WithStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />
          }
          label="Signets"
        />
      </WithStyles(FormControl)>
    </WithStyles(GridWrapper)>
    <WithStyles(GridWrapper)
      item={true}
      xs={12}
    >
      <WithStyles(Button)
        className={[MockFunction]}
        color="primary"
        onClick={[Function]}
        variant="raised"
      >
        Valider
      </WithStyles(Button)>
      <WithStyles(Button)
        className={[MockFunction]}
        color="secondary"
        onClick={[Function]}
        variant="raised"
      >
        Annuler
      </WithStyles(Button)>
    </WithStyles(GridWrapper)>
  </WithStyles(GridWrapper)>
</div>
`;
